src/google/protobuf/compiler/command_line_interface_unittest.cc

```diff
@@ -157,9 +153,11 @@ class CommandLineInterfaceTest : public testing::Test {
   void ExpectCapturedStdoutSubstringWithZeroReturnCode(
       const string& expected_substring);
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
```

Comment:
The one thing I'm worried about is that it's getting tough to follow all the ifdefs.

For this one, could you perhaps put an ifdef inside `HasAlternateErrorSubstring()` that makes it keep its current behavior if `defined(_WIN32) && !defined(__CYGWIN__)` is true, but otherwise return false? That way all the ifdef logic is hidden inside the function and most of the other ifdefs can disappear.